koku/api/settings/tag_management.py

```diff
@@ -0,0 +1,222 @@
+        sub_form_name = f"{SETTINGS_PREFIX}.tag_managment.subform"
+        sub_form_title = "Enable tags and labels"
+        sub_form_fields = [tag_key_text]
+        sub_form_fields.extend(components)
+        sub_form = create_subform(sub_form_name, sub_form_title, sub_form_fields)
+
+        return sub_form
+
+    def build_settings(self):
+        """
+        Generate OpenShift settings
+
+        Returns:
+            (List) - List of setting items
+        """
+        settings = self._build_tag_key()
+        return [settings]
+
+    def _tag_key_handler(self, settings):
+        """
+        Handle setting results
+
+        Args:
+            (String) name - unique name for switch.
+
+        Returns:
+            (Bool) - True, if a setting had an effect, False otherwise
+        """
+        updated = [False] * len(obtainTagKeysProvidersParams)
+        for ix, providerName in enumerate(obtainTagKeysProvidersParams):
+            enabled_tags = settings.get(providerName, {}).get("enabled", [])
+            remove_tags = []
+            tag_view = obtainTagKeysProvidersParams[providerName]["tag_view"]
+            query_handler = obtainTagKeysProvidersParams[providerName]["query_handler"]
+            enabled_tag_keys = obtainTagKeysProvidersParams[providerName]["enabled_tag_keys"]
+            provider = obtainTagKeysProvidersParams[providerName]["provider"]
+            available, _ = self._obtain_tag_keys(tag_view, query_handler, enabled_tag_keys)
+            invalid_keys = [tag_key for tag_key in enabled_tags if tag_key not in available]
+            if invalid_keys:
+                key = "settings"
+                message = f"Invalid tag keys provided: {', '.join(invalid_keys)}."
+                raise ValidationError(error_obj(key, message))
+            with schema_context(self.schema):
+                existing_enabled_tags = enabled_tag_keys.objects.all()
+                for existing_tag in existing_enabled_tags:
+                    if existing_tag.key in enabled_tags:
+                        enabled_tags.remove(existing_tag.key)
+                    else:
+                        remove_tags.append(existing_tag)
+                        updated[ix] = True
+                for rm_tag in remove_tags:
+                    rm_tag.delete()
+                for new_tag in enabled_tags:
+                    enabled_tag_keys.objects.create(key=new_tag)
+                    updated[ix] = True
+            if updated[ix]:
+                invalidate_view_cache_for_tenant_and_source_type(self.schema, provider)
+        return any(updated)
+
+    def handle_settings(self, settings):
+        """
+        Handle setting results
+
+        Args:
+            (String) name - unique name for switch.
+
+        Returns:
+            (Bool) - True, if a setting had an effect, False otherwise
+        """
+        openshift_settings = settings.get("api", {}).get("settings", {}).get("tag-management", {})
```

Comment:
`openshift_settings` is a left over, maybe something more general since it covers all source types now